src/main/java/be/seeseemelk/mockbukkit/entity/ShulkerMock.java

```diff
@@ -0,0 +1,100 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.DyeColor;
+import org.bukkit.Sound;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Shulker;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+/**
+ * Mock implementation of a {@link Shulker}.
+ *
+ * @see GolemMock
+ */
+public class ShulkerMock extends GolemMock implements Shulker
+{
+	private float peekAmount = 0;
+	private BlockFace attachedFace = BlockFace.DOWN;
+	private DyeColor dyeColor;
+
+	/**
+	 * Constructs a new {@link ShulkerMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public ShulkerMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public float getPeek()
+	{
+		return this.peekAmount;
+	}
+
+	@Override
+	public void setPeek(float value)
+	{
+		Preconditions.checkArgument(value >= 0 && value <= 1, "value needs to be in between or equal to 0 and 1");
+		this.peekAmount = value;
+	}
```

Comment:
I'm being notouriously picky this time:

The paper implementation trunkates the values to 1/100 values. (Hehe, the behavior will be slightly different if you write everything like that, i.e. `setPeek(0.3333333f)` should make `getPeek()` return 0.33f)

Reply:
Well spotted! I have update the code to match the correct behavior :)